transport/transport.go

```diff
@@ -81,6 +81,18 @@ type Transport interface {
 	// See the Network interface for an explanation of how this is used.
 	// TODO: Make this a part of the go-multiaddr protocol instead?
 	Proxy() bool
+	
+	// If your transport implements `io.Closer` Close will be called 
+	// when libp2p doesn't need it anymore.
+	// (libp2p closing, transport unhooked from a node, ...)
```

Comment:
Let's just simplify this to `when the libp2p node shuts down". There's no other way this will happen.